install_files/ansible-base/inventory-dynamic

```diff
@@ -0,0 +1,135 @@
+
+    except (IOError, KeyError):
+        msg = "You must run the configure playbook."
+        raise Exception(msg)
+
+    if hostname == "app":
+        return app_ip
+    elif hostname == "mon":
+        return monitor_ip
+    else:
+        msg = "Unsupported hostname: '{}'".format(hostname)
+        raise Exception(msg)
+
+
+def lookup_tor_hostname(hostname):
+    """
+    Extract Onion URL from HidServAuth file that was fetched back locally.
+    Returns Onion URL for given inventory hostname.
+    """
+    aths_path = os.path.join(SECUREDROP_ANSIBLE_DIRECTORY,
+                             "{}-ssh-aths".format(hostname))
+    with open(aths_path, 'r') as f:
+        tor_config = f.readline().rstrip().split()
+        try:
+            # Regardless of THS/ATHS, Onion URL will be first item.
+            # That's all we care about to get the address.
+            return tor_config[0]
+        except IndexError:
+            msg = ("Tor config file for '{}' ",
+                   "appears to be empty").format(hostname)
+            raise Exception(msg=msg)
+
+
+def lookup_ssh_address(hostname):
+    """
+    Wrapper function to first look up IPv4 address, then add ATHS
+    for Tor over SSH only if configured.
+    """
+    ssh_address = lookup_local_ipv4_address(hostname)
+    #import pdb; pdb.set_trace()
+    try:
+        ssh_address = lookup_tor_hostname(hostname)
+    # Don't assume ATHS files are present; they won't be on first run.
+    except (IndexError, EnvironmentError):
+        pass
+
+    return ssh_address
+
+
+def lookup_host_distro_id():
+    """
+    Returns output of `lsb_release --id --short`, to check
+    whether host machine is Tails.
+    """
+    cmd = ["lsb_release", "--id", "--short"]
+    p = subprocess.check_output(cmd)
+    distro_id = p.rstrip()
+    return distro_id
+
+
+def host_is_tails():
+    """
+    Wrapper return True/False for whether script is executed
+    within Tails. We don't want to add SSH extra args if True.
+    """
+    return lookup_host_distro_id() == "Tails"
+
+
+def build_inventory():
+    """
+    Construct JSON dict of host information for use as inventory.
+    """
+    inventory = {
+        "_meta": {
+            "hostvars": {
+                h: {
+                    "ansible_ssh_host": lookup_ssh_address(h)
+                } for h in SECUREDROP_SUPPORTED_HOSTNAMES
+            },
+        },
+        'securedrop': {'hosts': SECUREDROP_SUPPORTED_HOSTNAMES},
+        'securedrop_application_server': {'hosts': ['app']},
+        'securedrop_monitor_server': {'hosts': ['mon']},
+    }
+
+    # Append SSH extra args if NOT under Tails. Necessary for testing.
+    if not host_is_tails():
+        for host, hostvars in inventory["_meta"]["hostvars"].iteritems():
+            (inventory["_meta"]
+                      ["hostvars"]
+                      [host]
+                      ["ansible_ssh_common_args"]) = SSH_COMMON_ARGS
+    return inventory
+
+
+if __name__ == "__main__":
```

Comment:
Technically, there are a few things missing for this to be `compatible` with the ansible inventory spec. BUT it's working right now, so I'm not going to hold up the PR for this but FYI from the [docs](http://docs.ansible.com/ansible/dev_guide/developing_inventory.html) this script is missing:

* a `--list` option that well, does pretty much what its doing now
* a `--host <hostname>` option to return variables from a hostname or return an empty dict

( i also noticed our working infra inventory script for digitalocean is missing these) soo yeah :man_shrugging: